### PR TITLE
[RFC] unchanged: only use CHANGEDTICK if changed actually

### PR DESCRIPTION
**Pull request by blueyed**
Ref: https://github.com/vim/vim/issues/2764

### CODE REVIEW

File: src/buffer.c [hunk 1]
```diff
@@ -1410,6 +1410,8 @@ do_buffer(
 	if (bp == NULL && buf == curbuf)
 	    return empty_curbuf(TRUE, forceit, action);
 
+	++CHANGEDTICK(buf);
+
```
Comment (blueyed) [inline src/buffer.c:1414]: Note that this is just placed there rather arbitrary for Test_changedtick_bdel.

**Comment by blueyed** [comment]
Closing as per discussion in https://github.com/vim/vim/issues/2764.
I will just remove the workaround for BufWritePost with `:w` in Neomake, and people should use `:up` if they do not want checks to be re-run unnecessarily.

**Comment by brammool** [comment]
> blueyed commented on this pull request.
> 
> > @@ -1410,6 +1410,8 @@ do_buffer(
>  	if (bp == NULL && buf == curbuf)
>  	    return empty_curbuf(TRUE, forceit, action);
>  
> +	++CHANGEDTICK(buf);
> +
> 
> Note that this is just placed there rather arbitrary for
> Test_changedtick_bdel.

So, add a comment?

-- 
Amnesia is one of my favorite words, but I forgot what it means.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by blueyed** [comment]
> So, add a comment?

I've just put it there quickly to please the test - which I would have questioned anyway.  From looking quickly at the commit that added it it wasn't really clear why `:bdelete` has to change the tick.
I.e. I am pretty sure that there would be a better place/way to handle this probably - and just wanted to make this clear in the PR.

The issue is resolved for me anyway.

**Pull request closed without merging**